simplify-basedomain-usage/README.md

```diff
@@ -0,0 +1,186 @@
+Furthermore, if the value is part of the `Cluster` CR, it's the `Cluster` the one driving and triggering changes on cloud resources, allowing clusters to change or update their values independently of each other.
+This is a better scenario than our current approach having the controller receive this configuration as a parameter, because changing the parameter would trigger changes in cloud resources on all clusters.
+
+#### Reduce coupling
+
+Controllers fetching the `baseDomain` value from the `cluster-values` configmap depend on the `cluster-apps-operator` or `cluster-operator` deployed and creating the configmap.
+Using the value from the annotation instead of depending on the configmap reduces coupling between our controllers.
+
+#### Potential to use different DNS hosted zones on the same Management Cluster
+
+With our current approach, we configure the `baseDomain` at the installation level through the `config` repository, making the `baseDomain` shared for all clusters in a single installation.
+If we move the value to be part of the `Cluster` CR, different clusters could potentially use different hosted zones, even on the same Management Cluster.
+
+#### Migrate workload clusters to different management cluster
+
+One side effect from this is that by decoupling the `baseDomain` from the Management Cluster would also help moving clusters between installations, if that would be desired.
+
+
+### Downsides
+
+#### Migration
+
+For the migration to this approach, we would need to annotate all `Cluster` Custom Resources to have the annotation.
+We also would need to change our operators so that they read this value instead of the parameter they currently receive using the `config` repository.
+
+##### List of controllers that reconcile clusters and use the `baseDomain`
+
+###### aws-resolver-rules-operator
+
+It needs it to construct the k8s api endpoint of the cluster that's reconciling, so that it can create a Route53 Resolver Rule for it.
+
+###### cluster-apps-operator
+
+It needs it to store it in the `cluster-values` configmap that will be consumed by default apps of the cluster that's reconciling.
+
+###### dex-operator
+
+It needs it to configure dex for the cluster that's reconciling. The `baseDomain` passed to the app in the `config` repository is only used in `vintage` management clusters.
+For CAPI clusters or `vintage` workload clusters, the controller loads the `cluster-values` configmap from the management cluster.
+It should also take the value from the annotation instead.
+
+###### dns-operator-*
+
+It needs it to create the hosted zone and DNS records for the cluster that's reconciling.
+
+###### irsa-operator
+
+For CAPI clusters it loads the `cluster-values` configmap from the management cluster. It should also take the value from the annotation instead.
+
+## Proposal #2: Set value in default-apps-* config instead of using `baseDomain` from the `cluster-values` configmap in CAPI installations
```

Comment:
at the latest with the implementation of the `proxy` feature, the featureset of `cluster-apps-operator` is highly controversial. 

There is
* a [discussion in KaaS-sync](https://docs.google.com/document/d/15fBIUmv5pPvCF1DLwR6_XOaFLNHeVex8YNrVtgm83ME/edit#bookmark=id.j92bbrwm7zno)
* and [slack thread](https://gigantic.slack.com/archives/C01F7T2MNRL/p1669013425769809)
* which lead to a [github issue](https://github.com/giantswarm/giantswarm/issues/24784) to track that

when it came to a decision, we should consider/re-prioritize the `cluster-apps-operator` issue again.

> sorry for adding another complexity level here :hugs:

Reply:
[In this comment](https://github.com/giantswarm/giantswarm/issues/24784#issuecomment-1477348732) you wrote

> with the recent RFC regarding simplifying basedomain usage cluster-apps-operator got named as potential solution by extending the feature set again.

One of us is miss understanding something hehe I'm proposing to stop using the `baseDomain` value added by `cluster-apps-operator`. So it's not extending the feature set, but actually reducing it. I want to get rid of that value from the `cluster-values` configmap. Let me know if I miss understood your comment, or how I can improve the wording on the RFC, as it seems I didn't do a good job explaining it :smile:

Reply:
:see_no_evil: you're absolutely right ... i was to much focused on `cluster-apps-operator` that i totally misread your proposed solution.